### PR TITLE
Indiachan - New Presence

### PR DESCRIPTION
**Pull request by natto1784**
A small yet epic imageboard.

Proofs:

### CODE REVIEW

File: websites/I/Indiachan/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "sports",
+    "discussions",
+    "forum",
+    "community"
+  ],
+  "url": "indiachan.com",
+  "description": {
+        "hi-IN": "इंडियाचैन एक साधारण इमेजबोर्ड है जहां कोई भी किसी भी विषय के बारे में टिप्पणियां एवं चित्रों की सहायता से संवाद कर सकता है। आपको पोस्ट करने के लिए पंजीकरण करने की आवश्यकता नहीं है। इंडियाचैन विज्ञापनों या ट्रैकिंग का उपयोग नहीं करता है।",
+        "en": "Indiachan is a simple imageboard where anyone can post comments and share images about any topic. You do not need to register to post. Indiachan does not use ads or tracking. ",
+        "pa-IN": "ਇੰਡੀਆਚੈਨ ਇਕ ਸਧਾਰਨ ਚਿੱਤਰਬੋਰਡ ਹੈ ਜਿੱਥੇ ਕੋਈ ਵੀ ਕਿਸੇ ਵੀ ਵਿਸ਼ੇ ਬਾਰੇ ਗੱਲ-ਬਾਤ ਅਤੇ ਚਿੱਤਰ ਪੋਸਟ ਕਰ ਸਕਦਾ ਹੈ। ਤੁਹਾਨੂੰ ਪੋਸਟ ਕਰਨ ਲਈ ਰਜਿਸਟਰ ਕਰਨ ਦੀ ਲੋੜ ਨਹੀਂ ਹੈ। ਇੰਡੀਆਚੈਨ ਵਿਗਿਆਪਨ ਜਾਂ ਟਰੈਕਿੰਗ ਦੀ ਵਰਤੋਂ ਨਹੀਂ ਕਰਦਾ।"
```
Comment (Slowlife01) [inline websites/I/Indiachan/dist/metadata.json:27]:
```suggestion
        "hi_IN": "इंडियाचैन एक साधारण इमेजबोर्ड है जहां कोई भी किसी भी विषय के बारे में टिप्पणियां एवं चित्रों की सहायता से संवाद कर सकता है। आपको पोस्ट करने के लिए पंजीकरण करने की आवश्यकता नहीं है। इंडियाचैन विज्ञापनों या ट्रैकिंग का उपयोग नहीं करता है।",
        "en": "Indiachan is a simple imageboard where anyone can post comments and share images about any topic. You do not need to register to post. Indiachan does not use ads or tracking. "
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

**Comment by Slowlife01** [comment]
Also can you add more things?

**Comment by natto1784** [comment]
I've already added all that can be added, there isn't really much more to an imageboard. If I can add more stuff in the future, I'll add it.

**Comment by Slowlife01** [comment]
fix issues

**Comment by natto1784** [comment]
fixed indentation

**Comment by Slowlife01** [comment]
So is there more you can add?

**Comment by natto1784** [comment]
there isn't more i can add, I've added all that i could. Also, cleaned up previous commits for clarity

File: websites/I/Indiachan/presence.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const presence: Presence = new Presence({
+  clientId: "876813345699811328"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  }, p: string = document.location.pathname;
```
Comment (Slowlife01) [inline websites/I/Indiachan/presence.ts:8]:
```suggestion
  }, p = document.location.pathname;
```

File: websites/I/Indiachan/presence.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  if (p === "/") {
+    presenceData.details = "Home";
+    presenceData.smallImageText = "Idling";
+    presenceData.smallImageKey = "dindu";
+    presence.setActivity(presenceData);
+    return;
```
Comment (Slowlife01) [inline websites/I/Indiachan/presence.ts:14]: why set activity and return here

File: websites/I/Indiachan/presence.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    presenceData.smallImageKey = "dindu";
+  } else if (p.startsWith("/yoga")) {
+    presenceData.details = "Fitness";
+    presenceData.smallImageKey = "yoga";
+  } else if (p.startsWith("/dhan")) {
+    presenceData.details = "Dhandho";
+    presenceData.smallImageKey = "dhan";
+  }
+
+  const parts: string[] = p.split("/");
```
Comment (Slowlife01) [inline websites/I/Indiachan/presence.ts:45]:
```suggestion
  const parts = p.split("/");
```

File: websites/I/Indiachan/presence.ts (hunk 3 above)
Comment (natto1784) [reply websites/I/Indiachan/presence.ts:14]: to not make it go through the if else blocks below that

Comment (Slowlife01) [reply websites/I/Indiachan/presence.ts:14]: well it's not like I don't know that but you check if the pathname is "/" so why does the code below matter other than the browsing one which is small image text anyways

Comment (natto1784) [reply websites/I/Indiachan/presence.ts:14]: Because string functions like "includes" in the conditions are bound to make the code slower

Comment (Slowlife01) [reply websites/I/Indiachan/presence.ts:14]: well, to my experience I wouldn't say so but sure if you say so, I won't argue

Comment (natto1784) [reply websites/I/Indiachan/presence.ts:14]: Well I can just wrap the rest of the code in an else block if that's what you're asking me to do. But it won't be any different from using `return;` however.

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**